Library/Homebrew/version.rb

```diff
@@ -379,3 +392,29 @@ def self._parse(spec)
     return m.captures.first unless m.nil?
   end
 end
+
+class HeadVersion < Version
+  attr_reader :commit
+
+  def initialize(val)
+    if val.respond_to?(:to_str)
+      @version = val.to_str
+    else
+      raise TypeError, "Version value must be a string; got a #{val.class} (#{val})"
+    end
+    @commit = @version[/HEAD-(.+)/, 1]
+  end
```

Comment:
This is purely a style issue, but here's a suggestion that is easier to read in my opinion (similarly to another such comment I made earlier):

```ruby
def initialize(val)
  unless val.respond_to?(:to_str)
    raise TypeError, "Version value must be a string; got a #{val.class} (#{val})"
  end

  @version = val.to_str
  @commit = @version[/HEAD-(.+)/, 1]
end
```